src/wallet/rpcwallet.cpp

```diff
@@ -2749,6 +2749,12 @@ static RPCHelpMan createwallet()
         warnings.emplace_back(Untranslated("Wallet is an experimental descriptor wallet"));
     }
 
+#ifndef USE_BDB
```

Comment:
In commit "RPC: Require descriptors=True for createwallet when BDB is not compiled" (d2781aa656cce419700ba57a4551a999756750a4)

I think it would be good to drop the `#ifndef USE_BDB` here and the `#ifndef USE_SQLITE` a few lines above by just writing:

```c++
options.require_format = wallet_creation_flags & WALLET_FLAG_DESCRIPTORS ? DatabaseFormat::SQLITE : DatabaseFormat::BERKELEY;
```

in `CreateWallet`. It would be good for not having so many ifdefs, not having to hardcode the same database format assumption so many places, and having a separation of concerns where `MakeDatabase` callers don't need to figure out what kind of databases `MakeDatabase` can create before calling it

Reply:
This was done to have reasonable error messages. Can be cleaned up in a followup.